etna/search/views.py

```diff
@@ -526,49 +523,21 @@ def get_selected_filters(self, form: Form) -> Dict[str, List[Tuple[str, str]]]:
         if filter_keyword := form.cleaned_data.get("filter_keyword"):
             return_value.update({"filter_keyword": [(filter_keyword, filter_keyword)]})
 
-        # get form error messages
-        if error_dict := json.loads(form.errors.as_json()):
-            for dict_values in error_dict.values():
-                for item in dict_values:
-                    form_error_messages.append(item["message"])
-
         if opening_start_date := form.cleaned_data.get("opening_start_date"):
-            # if both dates have valid values but invalid when together
-            if (
-                CUSTOM_ERROR_MESSAGES.get("invalid_date_range")
-                not in form_error_messages
-            ):
```

Comment:
Because 'opening_start_date' will only be present in `form.cleaned_data` if that value is valid. There's no need to check for range validity explicitly here.

Reply:
When either of the date fields have errors, they are not in the selected field labels
When both of them (valid) but errors together only one of them show up in the selected filters. 
So the condition when removed allows one of them of show and not both, and when condition is put both don's show in filters labels - which is the same behaviour as if any one of them have errors.
Any thoughts on this?

Reply:
> When either of the date fields have errors, they are not in the selected field labels

In my view, it's fine for one or the other to show as a 'selected filter' (and be included to the API request) if it is valid but the other is not.

> When both of them (valid) but errors together only one of them show up in the selected filters.

Yep, that's still true. We make the 'start' date invalid, so only the 'end' date is shown as a selected filter